src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke.storage;
+
+import duke.exception.ErrorMessage;
+import duke.exception.StorageException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.util.TaskList;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a storage object that deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    public static final String DEFAULT_FILEPATH = "duke.txt";
+    private final Path filePath;
+
+    public Storage() {
+        this(DEFAULT_FILEPATH);
+    }
+
+    /**
+     * Creates a new storage with the specified file path.
+     *
+     * @param filePath Path of the file containing the stored tasks.
+     */
+    public Storage(String filePath) {
+        this.filePath = Paths.get(filePath);
+    }
```

Comment:
Good implementation! 

I did not consider checking for default/allowing users to decide where they want their data file to be stores,